tough-kms/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+mod client;
+pub mod error;
+use rusoto_kms::{Kms, KmsClient};
+use snafu::{OptionExt, ResultExt};
+use std::fmt;
+use tough::key_source::KeySource;
+use tough::schema::decoded::{Decoded, RsaPem};
+use tough::schema::key::KmsSigningAlgorithms;
+use tough::sign::{KmsRsaKey, Sign};
+
+/// Implements the KeySource trait for keys that live in AWS KMS
+pub struct KmsKeySource {
+    pub profile: Option<String>,
+    pub key_id: String,
+    pub client: Option<KmsClient>,
+}
+
+impl fmt::Debug for KmsKeySource {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("KmsKeySource")
+            .field("key_id", &self.key_id)
+            .field("profile", &self.profile)
+            .finish()
+    }
+}
+
+/// Implements the KeySource trait.
+impl KeySource for KmsKeySource {
+    fn as_sign(
+        &self,
+    ) -> std::result::Result<Box<dyn Sign>, Box<dyn std::error::Error + Send + Sync + 'static>>
+    {
+        let kms_client = match self.client.clone() {
+            Some(value) => value,
+            None => client::build_client_kms(self.profile.as_deref())?,
+        };
+        // Get the public key from aws kms
+        let fut = kms_client.get_public_key(rusoto_kms::GetPublicKeyRequest {
+            key_id: self.key_id.clone(),
+            ..rusoto_kms::GetPublicKeyRequest::default()
+        });
+        let response = tokio::runtime::Runtime::new()
+            .context(error::RuntimeCreation)?
+            .block_on(fut)
+            .context(error::KmsGetPublicKey {
+                profile: self.profile.clone(),
+                key_id: self.key_id.clone(),
+            })?;
+        let pb_key: Decoded<RsaPem> = response
+            .public_key
+            .context(error::PublicKeyNone)?
+            .to_vec()
+            .into();
+        Ok(Box::new(KmsRsaKey {
+            kms_client: kms_client.clone(),
+            key_id: self.key_id.clone(),
+            public_key: pb_key,
+            signing_algorithm: KmsSigningAlgorithms::Rsa(String::from("RSASSA_PSS_SHA_256")),
+        }))
+    }
+
+    fn write(
+        &self,
+        value: &str,
+    ) -> std::result::Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+        let kms_client = match self.client.clone() {
```

Comment:
See if this works (and self.pofile probably won't need as_deref() anymore?)

```suggestion
        let kms_client = match &self.client {
```